ui/src/app/views/recent-work/component.spec.ts

```diff
@@ -0,0 +1,157 @@
+import {ComponentFixture, fakeAsync, TestBed, tick} from '@angular/core/testing';
+
+import {By} from '@angular/platform-browser';
+import {ClarityModule} from '@clr/angular';
+import {Observable} from 'rxjs/Observable';
+
+import {RouterTestingModule} from '@angular/router/testing';
+
+import {FormsModule, ReactiveFormsModule} from '@angular/forms';
+
+import {CohortsService} from 'generated/api/cohorts.service';
+import {UserMetricsService} from 'generated/api/userMetrics.service';
+import {WorkspacesService} from 'generated/api/workspaces.service';
+
+import {CohortsServiceStub} from 'testing/stubs/cohort-service-stub';
+import {WorkspacesServiceStub} from 'testing/stubs/workspace-service-stub';
+
+import {simulateClick, updateAndTick} from 'testing/test-helpers';
+
+import {ResourceCardComponent} from 'app/views/resource-card/component';
+import {CohortEditModalComponent} from 'app/views/cohort-edit-modal/component';
+import {ConfirmDeleteModalComponent} from 'app/views/confirm-delete-modal/component';
+import {RecentWorkComponent} from 'app/views/recent-work/component';
+import {RenameModalComponent} from 'app/views/rename-modal/component';
+
+import {LeftScrollComponent} from 'app/icons/left-scroll/component';
+import {RightScrollComponent} from 'app/icons/right-scroll/component';
+
+describe('RecentWorkComponent', () => {
+  let fixture: ComponentFixture<RecentWorkComponent>;
+  let userMetricsSpy: jasmine.SpyObj<UserMetricsService>;
+  beforeEach(fakeAsync(() => {
+    const spy = jasmine.createSpyObj('UserMetricsService', ['getUserRecentResources']);
```

Comment:
In general I'd prefer fakes to mocks, which is what we're doing elsewhere for the other services.